.github/workflows/ci.yaml

```diff
@@ -23,3 +23,12 @@ jobs:
 
     - name: Run tests
       run: npm test
+
+    - name: Build and Push Docker image
+      env:
+        DOCKER_USERNAME: ${{ ayasaoudi }}  
```

Comment:
Don't hardcode your credentials. Put the values into [secret variables](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions).